VirtualH89/Src/z80.cpp

```diff
@@ -1533,37 +1538,42 @@ Z80::addClockTicks()
 }
 
 void
-Z80::setSpeed(bool fast)
-{
-    debugss(ssZ80, WARNING, "%s: - %d\n", __FUNCTION__, fast);
+Z80::setSpeedup(int factor) {
+    speedUpFactor_m = factor;
+}
 
-    if (fast)
-    {
+void
+Z80::enableFast() {
+    GppListener::addListener(this);
+}
+
+void
+Z80::gppNewValue(BYTE gpo) {
+    fast_m = ((gpo & z80_gppSpeedSelBit_c) != 0);
```

Comment:
The gppNewValue() is only called if the subscriber's "interested bits" change, so there's no need to check if fast_m actually changed - if this gets called you know it did.